### PR TITLE
Python 3 builtins via six

### PR DESCRIPTION
**Pull request by QuLogic**
Replace the builtins that are different on Python 3 using `six`. This only includes changes in files where these functions are used. I'll open a second PR adding these new builtins to the remaining files later.

### CODE REVIEW

File: lib/iris/tests/results/cdm/str_repr/simple.__str__.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
-thingness                                                                                (foo: 11)
+thingness                                                                                 (foo: 11)
      Dimension coordinates:
-          foo                                                                                x
+          foo                                                                                 x
      Auxiliary coordinates:
-          This is a really, really, really long long_name that requires to be clipped...     x
-          This is a short long_name                                                          x
\ No newline at end of file
+          This is a really, really, really, really long long_name that must be clipped...     x
```
Comment (pelson) [inline lib/iris/tests/results/cdm/str_repr/simple.__str__.txt]: curious....

File: lib/iris/tests/test_cdm.py [hunk 2]
```diff
@@ -338,9 +338,13 @@ def test_cube_summary_cell_methods(self):
     def test_cube_summary_alignment(self):
         # Test the cube summary dimension alignment and coord name clipping
         cube = iris.tests.stock.simple_1d()
-        aux = iris.coords.AuxCoord(range(11), long_name='This is a really, really, really long long_name that requires to be clipped because it is too long')
+        aux = iris.coords.AuxCoord(
+            np.arange(11),
+            long_name='This is a really, really, really, really long '
+                      'long_name that must be clipped because it is too long')
```
Comment (pelson) [inline lib/iris/tests/test_cdm.py]: explains it.

**Comment by pelson** [comment]
If tests pass :+1:. Please keep 3c9fc9f as a base commit, putting subsequent work on top of that.

**Comment by QuLogic** [comment]
OK, tests pass, but there might be one `filter` I missed somewhere that isn't caught right now. I'd prefer to have it in this PR instead of the next blanket one, so hold on for a bit.

**Comment by QuLogic** [comment]
Ah, I just missed the `.krb` file, so I think it should be good now (once tests pass).

**Comment by pelson** [comment]
Seems problematic...

**Comment by QuLogic** [comment]
Oops, I shouldn't do this so late at night...

**Pull request merged**